src/cobra/io/mat.py

```diff
@@ -232,18 +342,74 @@ def create_mat_dict(model: Model) -> OrderedDict:
     return mat
 
 
+def mat_annotations(
+    target_list: List[Object], mat_struct: np.ndarray, _d_replace: str = D_MET
+) -> None:
+    """Process mat structure annotations in place.
+
+    Will process mat structured annotations and add them to a list of new entities
+    (metabolites, reactions, genes) in a format based on identifiers.org.
+
+    Parameters
+    ----------
+    target_list: list[cobra.Object]
+        A list of cobra objects, including metabolites, reactions or genes. The
+        annotations will be added to these lists.
+    mat_struct: np.ndarray
+        A darray that includes the data imported from matlab file.
+    _d_replace: str
+        A string that points to the dictionary of converstions between MATLAB and
+        providers. Default D_MET (for metabolite).
+    """
+    annotation_matlab = list(
+        set(mat_struct.dtype.names).intersection(D_REPLACE[_d_replace].keys())
+    )
+    annotation_lists = [[None]] * len(annotation_matlab)
+    annotation_providers = [D_REPLACE[_d_replace][x] for x in annotation_matlab]
+    for i in range(len(annotation_matlab)):
+        annotation_lists[i] = _cell_to_str_list(mat_struct[annotation_matlab[i]][0, 0])
+        if annotation_matlab[i] == "rxnReferences":
+            # noinspection PyUnresolvedReferences
+            annotation_lists[i] = [
+                x.replace("PMID:", "") if x else None for x in annotation_lists[i]
+            ]
+        if annotation_matlab[i] == "rxnECNumbers":
+            # if there are more than one ec code, turn them to a comma separated str
+            # noinspection PyUnresolvedReferences,PyTypeChecker
+            annotation_lists[i] = [
+                ", ".join([y.strip() for y in x.split("or")])
+                if x and "or" in x
+                else None
+                for x in annotation_lists[i]
+            ]
+        if annotation_matlab[i] == "metCHEBIID":
+            # if there are more than one ec code, turn them to a comma separated str
+            # noinspection PyUnresolvedReferences,PyTypeChecker
+            annotation_lists[i] = [
+                x.replace("CHEBI:", "") if x else None for x in annotation_lists[i]
+            ]
+    for i in range(len(target_list)):
+        annotation_dict = {
+            annotation_providers[j]: annotation_lists[j][i]
+            for j in range(len(annotation_providers))
+            if annotation_lists[j][i]
+        }
+        if len(annotation_dict):
```

Comment:
Unnecessary, if the dict is empty you can still set it. The if probably generates more overhead.